modules/newrelic/widget.go

```diff
@@ -1,93 +1,69 @@
 package newrelic
 
 import (
-	"fmt"
+	"sort"
 
 	"github.com/rivo/tview"
 	"github.com/wtfutil/wtf/wtf"
-	nr "github.com/yfronto/newrelic"
 )
 
 type Widget struct {
+	wtf.KeyboardWidget
+	wtf.MultiSourceWidget
 	wtf.TextWidget
 
-	client   *Client
+	Clients []*Client
+
 	settings *Settings
 }
 
-func NewWidget(app *tview.Application, settings *Settings) *Widget {
+func NewWidget(app *tview.Application, pages *tview.Pages, settings *Settings) *Widget {
 	widget := Widget{
-		TextWidget: wtf.NewTextWidget(app, settings.common, false),
+		KeyboardWidget:    wtf.NewKeyboardWidget(app, pages, settings.common),
+		MultiSourceWidget: wtf.NewMultiSourceWidget(settings.common, "applicationID", "applicationIDs"),
+		TextWidget:        wtf.NewTextWidget(app, settings.common, true),
 
 		settings: settings,
 	}
 
-	widget.client = NewClient(widget.settings.apiKey, widget.settings.applicationID)
+	widget.initializeKeyboardControls()
+	widget.View.SetInputCapture(widget.InputCapture)
+
+	widget.SetDisplayFunction(widget.display)
+
+	widget.KeyboardWidget.SetView(widget.View)
 
 	return &widget
 }
 
 /* -------------------- Exported Functions -------------------- */
 
 func (widget *Widget) Refresh() {
-	app, appErr := widget.client.Application()
-	deploys, depErr := widget.client.Deployments()
-
-	appName := "error"
-	if appErr == nil {
-		appName = app.Name
+	for _, id := range wtf.ToInts(widget.settings.applicationIDs) {
+		widget.Clients = append(widget.Clients, NewClient(widget.settings.apiKey, id))
```

Comment:
This will create a new client that is calling the service each time, and this is probably not quite what you want. Say you had 3 applications, on the first refresh there would be 3, second would be 6, third would be 9..... eventually this will eat up a lot of resources.
It might make more sense to have the client accept applicationId as a parameter, and use that to iterate through requests and store the results in some struct, effectively caching the results. Otherwise, every time you change a page you will ALSO be making a web request to the newrelic API to get data, resulting in slower transitions. Minimally, I would create one set of clients up front, and Refresh will just call the display method